src/mds/CDir.h

```diff
@@ -226,6 +226,8 @@ class CDir : public MDSCacheObject, public Counter<CDir> {
   }
   fnode_t *project_fnode();
 
+  const utime_t& get_rstat_dirty_from();
```

Comment:
I don't see a reason for this to return a reference and it creates the danger of the reference dangling. Let's just return `utime_t`?